### DISCUSSION TITLE
Can't connect on server

### DISCUSSION THREAD

**Discussion by guarany-liv** [strapi-community/plugin-io#74] [Q&A]
I have a strapi 4.15.0 and installed the plugin. Set the plugins.ts to 
```
io: {
    enabled: true,
    config: {
      socket: {
        serverOptions: {
          cors: {
            origin: "*",
            methods: ["GET", "POST"]
          }
        }
      },
      contentTypes: [],
      events: [
        {
          name: "connection",
          handler({ strapi, io }, socket) {
            // will log whenever a socket connects
            strapi.log.info(
              `[io] new connection with id ${socket.id as number}`
            );
          }
        }
      ]
    }
  }
```

And I have a client for testing that is the same of the example on documentation. But I simply can't connect.
The configuration seems to be ok
![image](https://github.com/ComfortablyCoding/strapi-plugin-io/assets/143028023/e55b9ce9-620d-4bd1-a73d-30249e2a5851)

But I all get is: 
trying to connect http://localhost:4000
connect_error due to xhr poll error
connect_error due to xhr poll error
trying to connect http://localhost:4000
connect_error due to websocket error
connect_error due to websocket error

**Comment by guarany-liv** [comment]
I tried adding socketio to the bootstrap function without the plugin to see what happens, but the connection error also occurred.
```js
 bootstrap({ strapi }) {
    const io = new Server((strapi as LoadedStrapi).server.httpServer, {
      cors: {
        origin: "*",
        methods: ["GET", "POST"]
      }
    });

    io.on("connection", (socket) => {
      console.log("User connected");
    }
}
```

**Comment by ComfortablyCoding** [comment]
Versions 4.15.0 and 4.15.1 are deprecated version due to them having various issues. I would upgrade to the latest version (v4.15.5 ) and see if the issue still occurs.

**Comment by guarany-liv** [comment]
Hi! I tried updating the version but that didn't solve it, the problem kept happening. I also tried to create a new project from strapi 0 and the installation didn't work. I'll leave the link with some videos showing my tests.
I have 1 strapi project built which is on version 4.15.0. I tested it with a new strapi project in version 4.15.5 and it didn't work either. The example that worked was a socketio project with express to validate if the problem was the client.

https://drive.google.com/file/d/1AKCMV50tteT9r6U4T1HxMOfm5PpApjKR/view?usp=sharing

**Reply by ComfortablyCoding** [reply]
Strange, any chance you have your project in a public repo somewhere? If not is it possible to provide a minimal reproducible example?

**Comment by mr-mern** [comment]


**Reply by ComfortablyCoding** [reply]
"not working" is a fairly broad term, what exactly is not working? The connection? The events? Etc etc
